tests/components/sun/test_init.py

```diff
@@ -151,10 +137,10 @@
 
     assert dt_util.parse_datetime(
         state.attributes[sun.STATE_ATTR_NEXT_RISING]
-    ) == datetime(2016, 7, 25, 23, 23, 39, tzinfo=dt_util.UTC)
+    ) == datetime(2016, 7, 24, 22, 59, 45, 689645, tzinfo=timezone.utc)
```

Comment:
astral 2.1 returns these numbers now, double checked by using the library in a standalone test script, hope I'm not doing it wrong. Also the date parser returns datetime.timezone.utc, not pytz.utc, looks like ciso8601 does that.

Reply:
Unless the input to the test has changed the output shouldn't change. I think something is wrong.

Reply:
I agree, but you may verify these values using this very simple script: https://paste.xinu.at/rzpYfNPDbf3R2B9r/

Either the new astral is wrong and the tests should stay as is, or the old astral was wrong and the tests need to be adapted.

Reply:
It was up on my face the whole time and I didn't notice, but the sun rises before it sets, even though it is already up, so there's definitely sth wrong with the new astral.

Found an issue about it: https://github.com/sffjunkie/astral/issues/34

Reply:
The same issue has been there for a while, I had a look at fixing it ages ago but I didn't understand the maths involved. It's only in some polar regions that this happens.